ovs/lib/disk.py

```diff
@@ -98,9 +100,16 @@ def sync_with_reality(storagerouter_guid=None):
                             partition_name = device_name
                             partition_id = match.groups()[1]
                             device_name = match.groups()[0]
-                    sectors = int(client.run('cat /sys/block/{0}/size'.format(device_name)))
-                    sector_size = int(client.run('cat /sys/block/{0}/queue/hw_sector_size'.format(device_name)))
-                    rotational = int(client.run('cat /sys/block/{0}/queue/rotational'.format(device_name)))
+                    if device_name.startswith('md') and 'p' in device_name:
+                        md_device = device_name[:device_name.index('p')]
```

Comment:
If the 'is_partition is True' on line 82 is entered, the devicename will already become device_name = device_name[:device_name.index('p')]
Will it not screw up the check on line 104?